preprocess.py

```diff
@@ -0,0 +1,166 @@
+# Lint as: python3
+"Methods to preprocess data from different sources for text normalization."
+
+import re
+from typing import List
+from absl import app
+
+
+def process_data(data_file: str, data_source: str) -> List[str]:
+    """Processes data into list of strings depending on the data source.
+
+    Args:
+        data_file: The path to the data file.
+        data_source: Which corpus the data comes from.
+
+    Returns:
+        List of lines/sentences processed based on the data source.
+
+    Raises:
+        KeyError if you don't tell it which data source to use.
+    """
+    if data_source == "ud":
+        return process_ud_data(data_file)
+    elif data_source == "um":
+        return process_um_data(data_file)
+    elif data_source == "ac":
+        return process_ancrubadan_data(data_file)
+    elif data_source == "oscar":
+        return process_oscar_data(data_file)
+    elif data_source == "lcc":
+        return process_lcc_data(data_file)
+    print("Pick a data source!")
+    raise Exception
+
+
+def process_ud_data(ud_file: str) -> List[str]:
+    """Processes UD conllu file into list of strings for normalization.
+
+    UD data format follows the CONLL data format, where each sentence includes
+    metadata and each token is on a separate line. Each line includes info
+    about the token, lemma, part of speech, and dependency syntax. For this
+    function we only care about the token.
```

Comment:
An example line would help clarify this (and for the other data sources).

Reply:
I added some examples for each of the data sources. Some of them might need to be split across different lines though, since they exceed the maximum line length.